addons/account/models/account_account_tag.py

```diff
@@ -29,3 +29,9 @@ def _validate_tag_country(self):
         for record in self:
             if record.applicability == 'taxes' and not record.country_id:
                 raise ValidationError(_("A tag defined to be used on taxes must always have a country set."))
+
+    def unlink(self):
+        for rec in self:
+            if rec.get_external_id()[rec.id]:
+                raise UserError(_("You cannot delete %s tag as it is used in another module l10n_es_reports but you can archive it.") % rec.name)
```

Comment:
Manually created records may have external id if they were exported once.

Also, you shall not reference to `l10n_es_reports`, because the tag may belong to other module